server/provider/currenciesProvider.go

```diff
@@ -0,0 +1,61 @@
+package provider
+
+import "github.com/multiversx/mx-chain-rosetta/server/resources"
+
+type currenciesProvider struct {
+	nativeCurrency           resources.Currency
+	customCurrenciesSymbols  []string
+	customCurrencies         []resources.Currency
+	customCurrenciesBySymbol map[string]resources.Currency
+}
+
+// In the future, we might extract this to a standalone component (separate sub-package).
+// For the moment, we keep it as a simple structure, with unexported (future-to-be exported) member functions.
+func newCurrenciesProvider(nativeCurrencySymbol string, customCurrenciesSymbols []string) *currenciesProvider {
+	customCurrencies := make([]resources.Currency, 0, len(customCurrenciesSymbols))
+	customCurrenciesBySymbol := make(map[string]resources.Currency)
+
+	for _, symbol := range customCurrenciesSymbols {
+		customCurrency := resources.Currency{
+			Symbol: symbol,
+			// At the moment, for custom currencies (ESDTs), we hardcode the number of decimals to 0.
+			// In the future, we might fetch the actual number of decimals from the metachain observer.
+			Decimals: 0,
```

Comment:
wouldn't this cause overflows on 18 decimal tokens?
will the token supply/balance be expressed as uint64?

Reply:
Token supply / balance is expressed or handled as strings / big integers. The decimals aren't handled by Rosetta, they are simply forwarded to the HTTP client. Currently, hard-coded to 0, but now, after re-thinking, we can simply pass the number of decimals in the CLI, along with the tokens whitelist - no need for metachain observer :)